common/src/main/java/com/wynntils/wc/model/CharacterManager.java

```diff
@@ -136,5 +158,67 @@ public String toString() {
             return "CharacterInfo[classType=" + classType + ", reskinned=" + reskinned + ", level=" + level + ", id="
                     + id + ']';
         }
+
+        /**
+         * Return the maximum number of soul points the character can currently have
+         */
+        public int getMaxSoulPoints() {
+            // FIXME: If player is veteran, we should always return 15
+            int maxIfNotVeteran = 10 + MathUtils.clamp(getLevel() / 15, 0, 5);
+            if (getSoulPoints() > maxIfNotVeteran) {
+                return 15;
+            }
+            return maxIfNotVeteran;
+        }
+
+        /**
+         * Return the current number of soul points of the character, or -1 if unable to determine
+         */
+        public int getSoulPoints() {
+            ItemStack soulPoints = McUtils.player().getInventory().getItem(8);
+            if (soulPoints.getItem() != Items.NETHER_STAR) {
+                return -1;
+            }
+
+            return soulPoints.getCount();
+        }
+
+        /**
+         * Return the time in game ticks (1/20th of a second, 50ms) until the next soul point is given
+         *
+         * Also check that {@code {@link #getMaxSoulPoints()} >= {@link #getSoulPoints()}},
+         * in which case soul points are already full
+         */
+        public int getTicksToNextSoulPoint() {
+            if (McUtils.mc().level == null) return -1;
+            return 24000 - (int) (McUtils.mc().level.getDayTime() % 24000);
+        }
+
+        public int getTicksToNextManaRegen() {
```

Comment:
I think we should leave this out since we don't have the MR timer bar in anyway, and mana regen ticks will be removed in 2.0

Reply:
Or mark as deprecated

Reply:
Ok, I'll remove it. It is not used in this PR, and I just copied it over alongside the rest of the functions.